runtime/opt/zalando/init.d/10-prepare-disks.py

```diff
@@ -53,56 +83,134 @@ def is_mounted(mountpoint):
+    """Mounts formatted disks provided by /etc/taupage.yaml"""
+    if is_mounted is False and dir_exists is False:
+        subprocess.check_call(["mkdir", "-p", mountpoint])
+        subprocess.check_call(["mount", partition, mountpoint])
+    elif is_mounted is False and dir_exists is True:
+        subprocess.check_call(["mount", partition, mountpoint])
+    elif is_mounted is True and dir_exists is True:
+        logging.warning("Directory %s already exists and device is already mounted.", mountpoint)
+    else:
+        logging.error("Unexpected error while mounting the disks")
 
 
 def iterate_mounts(config):
-    '''Iterates over mount points file to provide disk device paths'''
-    for mpoint, data in config.get("mounts", {}).items():
+    """Iterates over mount points file to provide disk device paths"""
+    for mountpoint, data in config.get("mounts", {}).items():
         # mount path below /mounts on the host system
         # (the path specifies the mount point inside the Docker container)
-        mpoint = '/mounts/{}'.format(mpoint)
-        format_disks(mpoint, data['devices'], data.get("erase_on_boot", False), data.get("filesystem", "ext4"), is_mounted(mpoint))
-        mount_disks(mpoint, data['devices'], dir_exists(mpoint), is_mounted(mpoint))
-        subprocess.check_call(["chown", "-R", "application:", mpoint])
+        mountpoint = '/mounts/{}'.format(mountpoint)
 
+        partition = data.get("partition")
+        filesystem = data.get("filesystem", "ext4")
+        initialize = data.get("erase_on_boot", False)
+        already_mounted = is_mounted(mountpoint)
+
+        format_partition(partition, filesystem, initialize, already_mounted, config.get('root'))
+        mount_partition(partition, mountpoint, dir_exists(mountpoint), already_mounted)
 
-def main():
 
+def handle_ebs_volumes(args, ebs_volumes):
+    current_region = args.region if args.region else region()
+    ec2 = boto.ec2.connect_to_region(current_region)
+    for device, name in ebs_volumes.items():
+        attach_volume(ec2, find_volume(ec2, name), device)
+
+
+def raid_device_exists(raid_device):
+    try:
+        subprocess.check_call(["mdadm", raid_device])
+        return True
+    except:
+        return False
+
+
+def create_raid_device(raid_device, raid_config):
+    devices = raid_config.get("devices", [])
+    num_devices = len(devices)
+    if num_devices < 2:
+        logging.error("You need at least 2 devices to create a RAID")
+        sys.exit(4)
+    else:
+        call = ["mdadm",
+                "--build", raid_device,
+                "--level=" + str(raid_config.get("level")),
+                "--raid-devices=" + str(num_devices)]
+        for device in devices:
+            tries = 0
+            # Give devices some time to be available in case they were recently attached
+            while tries < 3 and not os.path.exists(device):
+                logging.error("Waiting for %s to stabilize", device)
```

Comment:
can you configure "logging" and use the appropriate log level? (I guess INFO would be fine here)